File: backend/app/schemas/meal.py
from pydantic import BaseModel, ConfigDict


class MealCreate(BaseModel):
    date: str
    breakfast: bool
    lunch: bool
    dinner: bool


class MealResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    date: str
    breakfast: bool
    lunch: bool
    dinner: bool
    user_id: int


class MealSummaryResponse(BaseModel):
    total_meal_slots: int

File: types/meal.ts
export type Meal = {
  id: number;
  date: string;
  breakfast: boolean;
  lunch: boolean;
  dinner: boolean;
  user_id: number;
};

export type MealCreate = {
  date: string;
  breakfast: boolean;
  lunch: boolean;
  dinner: boolean;
};

export type MealSummary = {
  total_meal_slots: number;
};
